pattern/variable/TypeVariable.java

```diff
@@ -254,11 +255,11 @@ public void addTo(GraphTraversal.Thing traversal) {
     }
 
     @Override
-    public AlphaEquivalence alphaEquals(TypeVariable that) {
-        return AlphaEquivalence.valid()
-                .validIf(id().isName() == that.id().isName())
-                .validIfAlphaEqual(labelConstraint, that.labelConstraint)
-                .validIfAlphaEqual(valueTypeConstraint, that.valueTypeConstraint)
-                .addMapping(this, that);
+    public FunctionalIterator<AlphaEquivalence> alphaEquals(TypeVariable that) {
+        assert !inferredTypes().isEmpty();
+        return AlphaEquivalence.empty()
+                .alphaEqualIf(id().isName() == that.id().isName())
+                .flatMap(a -> a.alphaEqualIf(inferredTypes().equals(that.inferredTypes())))
+                .map(a -> a.extend(this, that));
```

Comment:
Given our new idea of what Alpha Equivalence does for Variable/Constraint/Concludable, at this level we can I think generalise to handle named and labeled variables - as long as they are the same type of identifier, it's ok.

Reply:
I think this is the same problem as above, we're already doing it!